Add binary_sensor platform to LG ThinQ integration

LG-ThinQ-Integration opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Add binary_sensor platform to LG ThinQ integration


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/125054
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **New Features**
	- Added support for binary sensors in the LG ThinQ integration, enhancing device management capabilities.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

LG-ThinQ-Integration committed:
Add binary_sensor platform to LG ThinQ integration

c0ffeeca7 committed:
Merge branch 'next' into next

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -11,6 +11,7 @@ ha_config_flow: true
 ha_domain: lg_thinq
 ha_platforms:
   - switch
+  - binary_sensor
```
between Supported devices and Troubleshooting, please add a Section ## Binary sensors listing the names and function of all supported binary sensors

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -75,6 +76,39 @@ Support LG Appliances as follows:
+| [Refrigerator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/refrigerator/) | `ecoFriendly` | `ecoFriendlyMode` |
+| | `powerSave` | `powerSaveEnabled` |
+| | `sabbath` | `sabbathMode` |
+| [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Wine-Cellar/) | `operation` | `sabbathMode` |
```
```suggestion
| [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Wine-Cellar/) | `operation` | `Sabbath mode` |
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -75,6 +76,39 @@ Support LG Appliances as follows:
+| [Dish Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/dish-washer/) | `dishWashingStatus` | `rinseRefill` |
+| [Refrigerator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/refrigerator/) | `ecoFriendly` | `ecoFriendlyMode` |
+| | `powerSave` | `powerSaveEnabled` |
+| | `sabbath` | `sabbathMode` |
```
```suggestion
| | `sabbath` | Sabbath mode |
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -75,6 +76,39 @@ Support LG Appliances as follows:
+| [Cooktop](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Cooktop/)<br>[Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Dryer/)<br>[Oven](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Oven/)<br>[Styler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/styler/)<br>Washcombo Main<br ...
+| [Dish Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/dish-washer/) | `dishWashingStatus` | `rinseRefill` |
+| [Refrigerator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/refrigerator/) | `ecoFriendly` | `ecoFriendlyMode` |
+| | `powerSave` | `powerSaveEnabled` |
```
```suggestion
| | `powerSave` | Power saving mode |
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -75,6 +76,39 @@ Support LG Appliances as follows:
+| ------ | -------- | -------- |
+| [Cooktop](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Cooktop/)<br>[Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Dryer/)<br>[Oven](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Oven/)<br>[Styler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/styler/)<br>Washcombo Main<br ...
+| [Dish Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/dish-washer/) | `dishWashingStatus` | `rinseRefill` |
+| [Refrigerator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/refrigerator/) | `ecoFriendly` | `ecoFriendlyMode` |
```
```suggestion
| [Refrigerator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/refrigerator/) | `ecoFriendly` | Eco-friendly mode |
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -75,6 +76,39 @@ Support LG Appliances as follows:
+| Device | Resource | Property |
+| ------ | -------- | -------- |
+| [Cooktop](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Cooktop/)<br>[Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Dryer/)<br>[Oven](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Oven/)<br>[Styler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/styler/)<br>Washcombo Main<br ...
+| [Dish Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/dish-washer/) | `dishWashingStatus` | `rinseRefill` |
```
```suggestion
| [Dish Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/dish-washer/) | `dishWashingStatus` | Rinse refill needed |
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -75,6 +76,39 @@ Support LG Appliances as follows:
+
+| Device | Resource | Property |
+| ------ | -------- | -------- |
+| [Cooktop](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Cooktop/)<br>[Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Dryer/)<br>[Oven](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Oven/)<br>[Styler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/styler/)<br>Washcombo Main<br ...
```
```suggestion
| [Cooktop](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Cooktop/)<br>[Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Dryer/)<br>[Oven](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Oven/)<br>[Styler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/styler/)<br>Washcombo Main<br>Washcombo Mini<br>[Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/washer/)<br>[Washtower Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Dryer/)<br>[Washtower](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Single-Unit/)<br>[Washtower Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Washer/)<br> | `remoteControlEnable` | Enable remote control |
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -75,6 +76,39 @@ Support LG Appliances as follows:
 {% endnote %}
 
+## Platforms
+
```
Nice! Thank you for adding those! 
- Instead of listing the technical property name, I think we could use the user-facing string. Please correct me if I got those suggestions wrong.
- Not sure if the resource columns is needed?
- Could you add a column that provides a short description of the functionality? For example, it is not clear to me what the Sabbath mode does or what the difference is between eco-friendly and power-saving?

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

c0ffeeca7 commented:
I'm [drafting this PR](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed), as changes are needed. Please undraft it, once the changes have been implemented and the PR is ready for review again.

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration committed:
Update lg_thinq.markdown

LG-ThinQ-Integration commented:
> I'm [drafting this PR](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed), as changes are needed. Please undraft it, once the changes have been implemented and the PR is ready for review again.

I guess you missed my comments above.
1. I changed property as you mentioned.
2. I removed resource column as you mentioned.
3. I didn't add description because we don't need them here. There the links for them. And it is quite simple so that a user can guess from property column. And the thing you mentioned is a religious thing. So we can't add any description for that based on Home Assistant PR Guide.

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -75,6 +76,39 @@ Support LG Appliances as follows:
+| [Dish Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/dish-washer/) | Rinse refill needed |
+| [Refrigerator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/refrigerator/) | Eco-friendly mode |
+| | Power saving mode  |
+| | Enable sabbath mode |
```
```suggestion
| | Enable Sabbath mode |
```

c0ffeeca7 commented on source/_integrations/lg_thinq.markdown:
```diff
@@ -75,6 +76,39 @@ Support LG Appliances as follows:
+| [Refrigerator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/refrigerator/) | Eco-friendly mode |
+| | Power saving mode  |
+| | Enable sabbath mode |
+| [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Wine-Cellar/) | Enable sabbath mode |
```
```suggestion
| [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Wine-Cellar/) | Enable Sabbath mode |
```

c0ffeeca7 committed:
tweak

c0ffeeca7 reviewed (approved):
Thank you, @LG-ThinQ-Integration :+1:
